src/elements/content-explorer/__tests__/ContentExplorer.test.js

```diff
@@ -559,6 +559,50 @@ describe('elements/content-explorer/ContentExplorer', () => {
         });
     });
 
+    describe('handleSharedLinkSuccess()', () => {
+        const getApiShareMock = jest.fn().mockImplementation((newItem, access, callback) => callback());
+        const getApiMock = jest.fn().mockReturnValue({ share: getApiShareMock });
+        const updateCollectionMock = jest.fn();
+
+        const boxItem = {
+            shared_link: 'not null',
+            permissions: 'not null',
+            type: 'not null',
+        };
+
+        let wrapper;
+        let instance;
+
+        beforeEach(() => {
+            wrapper = getWrapper({ canShare: true, canSetShareAccess: true });
+            instance = wrapper.instance();
+            instance.api = { getAPI: getApiMock };
+            instance.updateCollection = updateCollectionMock;
+        });
+
+        afterEach(() => {
+            getApiShareMock.mockClear();
+            getApiMock.mockClear();
+            updateCollectionMock.mockClear();
+        });
+
+        test('should create shared link when it doesnt exist yet', () => {
+            instance.handleSharedLinkSuccess({ ...boxItem, shared_link: null });
```

Comment:
I understand testing like this is an anti-pattern, but invoking this function naturally requires a very large amount of setup

Reply:
These are unit tests and not integration tests so I don't mind favoring simplicity here as long as the tests are comprehensive. So I think this is ok.